src/Microsoft.Net.Http.Headers/MediaTypeHeaderValue.cs

```diff
@@ -0,0 +1,408 @@
+            get
+            {
+                var charset = Charset;
+                if (!string.IsNullOrWhiteSpace(charset))
+                {
+                    try
+                    {
+                        return Encoding.GetEncoding(charset);
+                    }
+                    catch (ArgumentException)
+                    {
+                        // Invalid or not supported
+                    }
+                }
+                return null;
+            }
+            set
+            {
+                if (value == null)
+                {
+                    Charset = null;
+                }
+                else
+                {
+                    Charset = value.WebName;
+                }
+            }
+        }
+
+        public string Boundary
+        {
+            get
+            {
+                return NameValueHeaderValue.Find(_parameters, BoundaryString)?.Value;
+            }
+            set
+            {
+                var boundaryParameter = NameValueHeaderValue.Find(_parameters, BoundaryString);
+                if (string.IsNullOrEmpty(value))
+                {
+                    // Remove charset parameter
+                    if (boundaryParameter != null)
+                    {
+                        _parameters.Remove(boundaryParameter);
+                    }
+                }
+                else
+                {
+                    if (boundaryParameter != null)
+                    {
+                        boundaryParameter.Value = value;
+                    }
+                    else
+                    {
+                        Parameters.Add(new NameValueHeaderValue(BoundaryString, value));
+                    }
+                }
+            }
+        }
+
+        public ICollection<NameValueHeaderValue> Parameters
+        {
+            get
+            {
+                if (_parameters == null)
+                {
+                    _parameters = new ObjectCollection<NameValueHeaderValue>();
+                }
+                return _parameters;
+            }
+        }
+
+        public double? Quality
+        {
+            get { return HeaderUtilities.GetQuality(Parameters); }
+            set { HeaderUtilities.SetQuality(Parameters, value); }
+        }
+
+        public string MediaType
+        {
+            get { return _mediaType; }
+            set
+            {
+                CheckMediaTypeFormat(value, "value");
+                _mediaType = value;
+            }
+        }
+
+        public string Type
+        {
+            get
+            {
+                return _mediaType.Substring(0, _mediaType.IndexOf('/'));
+            }
+        }
+
+        public string SubType
+        {
+            get
+            {
+                return _mediaType.Substring(_mediaType.IndexOf('/') + 1);
+            }
+        }
+
+        /// <summary>
+        /// MediaType = "*/*"
+        /// </summary>
+        public bool AllTypes
```

Comment:
Recommendation: should be "IsAllTypes" or similar since you're asking "is this X?"  Or "Matches", "Supports", etc.